recOrder/cli/parsing.py

```diff
@@ -1,54 +1,76 @@
-import click
+from pathlib import Path
 from typing import Callable
-from iohub.ngff import open_ome_zarr, Plate
+
+import click
+from iohub.ngff import Plate, open_ome_zarr
+from natsort import natsorted
+
+from recOrder.cli.option_eat_all import OptionEatAll
 
 
-def _validate_fov_path(
+def _validate_and_process_paths(
     ctx: click.Context, opt: click.Option, value: str
 ) -> None:
-    dataset = open_ome_zarr(value)
-    if isinstance(dataset, Plate):
-        raise ValueError(
-            "Please supply a single position instead of an HCS plate. Likely fix: replace 'input.zarr' with 'input.zarr/0/0/0'"
-        )
-    return value
+    # Sort and validate the input paths
+    input_paths = [Path(path) for path in natsorted(value)]
+    for path in input_paths:
+        with open_ome_zarr(path, mode="r") as dataset:
+            if isinstance(dataset, Plate):
+                raise ValueError(
+                    "Please supply a list of positions instead of an HCS plate. Likely fix: replace 'input.zarr' with 'input.zarr/*/*/*' or 'input.zarr/0/0/0'"
+                )
+    return input_paths
 
 
-def input_data_path_argument() -> Callable:
+def input_position_dirpaths() -> Callable:
     def decorator(f: Callable) -> Callable:
-        return click.argument(
-            "input-data-path",
-            type=click.Path(exists=True),
-            callback=_validate_fov_path,
-            nargs=1,
+        return click.option(
+            "--input-position-dirpaths",
+            "-i",
+            cls=OptionEatAll,
+            type=tuple,
+            required=True,
+            callback=_validate_and_process_paths,
+            help="List of paths to input positions, each with the same TCZYX shape. Supports wildcards e.g. 'input.zarr/*/*/*'.",
```

Comment:
@ziw-liu here's the improved docstring...can add elsewhere too.

Reply:
Ah I didn't find this because I was expecting 'transfer function computation assumes that input datasets all have the same shape'

Reply:
Good point, I just improved the docstrings and README.